src/main/java/edu/harvard/iq/dataverse/UserServiceBean.java

```diff
@@ -42,7 +50,459 @@ public AuthenticatedUser save(AuthenticatedUser user) {
+                        if (!userRoleList.contains(groupRole)){
+                            //System.out.println("User Role: "+groupRole);
+                            userRoleList.add(groupRole);
+                            userRoleLookup.put(groupUserIdentifier, userRoleList);
+                        }
+                    }
+                }
+            }
+        }
+        
+        
+        return userRoleLookup;
+    }
+    
+    /**
+     * 
+     * @param userId
+     * @return 
+     */
+    private String getUserRolesAsString(Integer userId) {
+        String retval = "";
+        String userIdentifier = "";
+        String qstr = "select useridentifier ";
+        qstr += " FROM authenticateduser";
+        qstr += " WHERE id = " + userId.toString();
+        qstr += ";";
+        
+        Query nativeQuery = em.createNativeQuery(qstr);
+
+        userIdentifier = '@' + (String) nativeQuery.getSingleResult();
+
+        qstr = " select distinct d.name from roleassignment a, dataverserole d";
+        qstr += " where d.id = a.role_id and a.assigneeidentifier='" + userIdentifier + "'"
+                + " Order by d.name;";
+
+        nativeQuery = em.createNativeQuery(qstr);
+
+        List<Object[]> roleList = nativeQuery.getResultList();
+
+        for (Object o : roleList) {
+            if (!retval.isEmpty()) {
+                retval += ", ";
+            }
+            retval += (String) o;
+        }
+        return retval;
+    }
+    
+    /**
+     * 
+     * Run a native query, returning a List<Object[]> containing
+     * AuthenticatedUser information as well as information about the 
+     * Authenticated Provider (e.g. builtin user, etc)
+     * 
+     * @param searchTerm
+     * @param sortKey
+     * @param resultLimit
+     * @return 
+     */
+    private List<Object[]> getUserListCore(String searchTerm, String sortKey, Integer resultLimit, Integer offset) {
+
+        if ((sortKey == null) || (sortKey.isEmpty())){
+            sortKey = "u.username";
+        }else{
+            sortKey = "u." + sortKey;
+        }
+        
+        if ((resultLimit == null)||(resultLimit < 1)){
+            resultLimit = 1;
+        }
+        
+        if ((searchTerm==null)||(searchTerm.isEmpty())){
+            searchTerm = "";
+        }
+        
+        if ((offset == null)||(offset < 0)){
+            offset = 0;
+        }
+        
+        //Results of this query are used to build Authenticated User records:
+
+        searchTerm = searchTerm.trim();
+
+        String sharedSearchClause = "";
+        
+        if (!searchTerm.isEmpty()) {
+            sharedSearchClause = " AND " + getSharedSearchClause();
+        }
+        
+        
+        String qstr = "SELECT u.id, u.useridentifier,";
+        qstr += " u.lastname, u.firstname, u.email,";
+        qstr += " u.affiliation, u.superuser,";
+        qstr += " u.position, u.modificationtime,";
```

Comment:
I'm on 375e4b8 and I'm getting `PSQLException: ERROR: column u.modificationtime does not exist`. This field was removed in 20df0e6. In my browser I'm seeing "Internal Server Error" at http://localhost:8080/dashboard-users.xhtml